reports/tests.py

```diff
@@ -314,21 +322,22 @@ def midnight(self, timestamp):
                                  tzinfo=pytz.timezone(settings.TIME_ZONE))
 
     def trigger_report_generation(self):
-        tmp_file = self.mk_tempfile()
+        with mock.patch('random.choice', mockobj.choice):
+            filename = generate_random_filename()
 
-        generate_report.apply_async(kwargs={
-            'start_date': self.midnight(datetime.strptime('2016-01-01',
-                                                          '%Y-%m-%d')),
-            'end_date': self.midnight(datetime.strptime('2016-02-01',
-                                                        '%Y-%m-%d')),
-            'output_file': tmp_file.name,
-            'email_recipients': ['foo@example.com'],
-            'email_subject': 'The Email Subject'})
+            generate_report.apply_async(kwargs={
+                'start_date': self.midnight(datetime.strptime('2016-01-01',
+                                                              '%Y-%m-%d')),
+                'end_date': self.midnight(datetime.strptime('2016-02-01',
+                                                            '%Y-%m-%d')),
+                'email_recipients': ['foo@example.com'],
+                'email_subject': 'The Email Subject'})
 
-        return tmp_file
+            return filename
 
     @responses.activate
-    def test_generate_report_email(self):
+    @mock.patch("os.remove")
```

Comment:
if we're mocking the removal in all these tests won't we end up with a bunch of files that aren't deleted at the end of the tests?